.github/workflows/main.yml

```diff
@@ -107,6 +113,66 @@ jobs:
           retention-days: 1
           if-no-files-found: error
 
+      - name: Build all Tests
+        if: env.test_tests == 'true' && success()
+        run: ant -quiet -Dcluster.config=release test -Dtest.includes=NoTestsJustBuild
+
+
+  java-cluster:
+    name: Build Java Cluster on JDK ${{ matrix.java }}
+    if: contains(github.event.pull_request.labels.*.name, 'ci:no-build') == false
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '11' ]
+      fail-fast: false
+    steps:
+        
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - name: Setup Xvfb
+        if: ${{ matrix.java != '20-ea' }}  # see #4299
+        run: |
+          echo "DISPLAY=:99.0" >> $GITHUB_ENV
+          Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+
+      - name: Checkout ${{ github.ref }} ( ${{ github.sha }} )
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          submodules: false
+
+      - name: Caching dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('*/external/binaries-list', '*/*/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Build Java Cluster
+        run: |
+          ant $OPTS -quiet -Dcluster.config=java -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false clean
+          ant $OPTS -quiet -Dcluster.config=java -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false build
+        
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: 'zulu'
+
+      - name: java/java.lsp.server
+        run: ant $OPTS -Dcluster.config=java -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -f java/java.lsp.server test
```

Comment:
I am having difficulties migrating the `java/java.lsp.server` job. Its an unreliable job on travis which makes it difficult to figure out what is wrong even if it reliably fails, since at least half of the failures are the sporadic failures.

I tried to replicate similar conditions as on travis, I even detached it from the pipeline and build the `java` cluster just for this test on JDK 11, then switch to 8 and run the test.

this error looks suspicious since I couldn't find it in sporadic local failures so far (I hope this isn't just yet another sporadic one):
```
    [junit] Oct 30, 2022 10:30:08 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
    [junit] SEVERE: Internal error: java.lang.NoClassDefFoundError: org/netbeans/modules/java/lsp/server/htmlui/$JsCallbacks$
    [junit] java.util.concurrent.CompletionException: java.lang.NoClassDefFoundError: org/netbeans/modules/java/lsp/server/htmlui/$JsCallbacks$
    [junit] 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
    [junit] 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
    [junit] 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1606)
    [junit] 	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
    [junit] 	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
    [junit] 	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
    [junit] 	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
    [junit] Caused by: java.lang.NoClassDefFoundError: org/netbeans/modules/java/lsp/server/htmlui/$JsCallbacks$
    [junit] 	at org.netbeans.modules.java.lsp.server.htmlui.Buttons.createButton0(Buttons.java)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer.createButton(AbstractLspHtmlViewer.java:56)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer.createButton(AbstractLspHtmlViewer.java:41)
    [junit] 	at org.netbeans.modules.htmlui.Buttons.createButton(Buttons.java:150)
    [junit] 	at org.netbeans.modules.htmlui.Buttons.buttons(Buttons.java:107)
    [junit] 	at org.netbeans.modules.htmlui.HTMLDialogBase$1AcceptAndInit.initializeButtons(HTMLDialogBase.java:86)
    [junit] 	at org.netbeans.modules.htmlui.HTMLDialogBase$1AcceptAndInit.call(HTMLDialogBase.java:78)
    [junit] 	at org.netbeans.modules.htmlui.HTMLDialogBase$1AcceptAndInit.call(HTMLDialogBase.java:46)
    [junit] 	at org.netbeans.spi.htmlui.HTMLViewerSpi$Context.onPageLoad(HTMLViewerSpi.java:168)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.MockHtmlViewer.lambda$load$0(MockHtmlViewer.java:52)
    [junit] 	at net.java.html.BrwsrCtx$1Wrap.run(BrwsrCtx.java:120)
    [junit] 	at net.java.html.BrwsrCtx.execute(BrwsrCtx.java:129)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.MockHtmlViewer.load(MockHtmlViewer.java:49)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.MockHtmlViewer.component(MockHtmlViewer.java:73)
    [junit] 	at org.netbeans.modules.java.lsp.server.ui.MockHtmlViewer.component(MockHtmlViewer.java:40)
    [junit] 	at org.netbeans.modules.htmlui.HtmlPair.component(HtmlPair.java:64)
    [junit] 	at org.netbeans.modules.htmlui.HTMLDialogBase.create(HTMLDialogBase.java:96)
    [junit] 	at org.netbeans.api.htmlui.HTMLDialog$Builder.show(HTMLDialog.java:227)
    [junit] 	at org.netbeans.modules.java.lsp.server.refactoring.Pages.showMoveMembersUI(Pages.java:63)
    [junit] 	at org.netbeans.modules.java.lsp.server.refactoring.MoveRefactoring.lambda$null$0(MoveRefactoring.java:164)
    [junit] 	at org.netbeans.api.java.source.JavaSource$MultiTask.run(JavaSource.java:504)
    [junit] 	at org.netbeans.modules.parsing.impl.TaskProcessor.callUserTask(TaskProcessor.java:586)
    [junit] 	at org.netbeans.modules.parsing.api.ParserManager$UserTaskAction.run(ParserManager.java:132)
    [junit] 	at org.netbeans.modules.parsing.api.ParserManager$UserTaskAction.run(ParserManager.java:116)
    [junit] 	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:181)
    [junit] 	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:178)
    [junit] 	at org.netbeans.modules.masterfs.filebasedfs.utils.FileChangedManager.priorityIO(FileChangedManager.java:153)
    [junit] 	at org.netbeans.modules.masterfs.providers.ProvidedExtensions.priorityIO(ProvidedExtensions.java:335)
    [junit] 	at org.netbeans.modules.parsing.nb.DataObjectEnvFactory.runPriorityIO(DataObjectEnvFactory.java:118)
    [junit] 	at org.netbeans.modules.parsing.impl.Utilities.runPriorityIO(Utilities.java:67)
    [junit] 	at org.netbeans.modules.parsing.impl.TaskProcessor.runUserTask(TaskProcessor.java:178)
    [junit] 	at org.netbeans.modules.parsing.api.ParserManager.parse(ParserManager.java:83)
    [junit] 	at org.netbeans.api.java.source.JavaSource.runUserActionTaskImpl(JavaSource.java:454)
    [junit] 	at org.netbeans.api.java.source.JavaSource.runUserActionTask(JavaSource.java:425)
    [junit] 	at org.netbeans.modules.java.lsp.server.refactoring.MoveRefactoring.lambda$processCommand$1(MoveRefactoring.java:156)
    [junit] 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604)
    [junit] 	... 4 more
```

@sdedic does this error look familiar to you or is this a new one?